Gemfile

```diff
@@ -252,7 +252,13 @@ end
 def override_gem(name, *args)
   if dependencies.any?
     raise "Trying to override unknown gem #{name}" unless (dependency = dependencies.find { |d| d.name == name })
-    dependencies.delete(dependency)
+
+    removed_dependency = dependencies.delete(dependency)
+    if removed_dependency.source.kind_of?(Bundler::Source::Git)
```

Comment:
@Fryguy Well, since the `Gemfile` (and sub-`Gemfiles`, aka `bundler.d/*.rb`) are evaluated in serial, if the `git` gem showed up first, when the `override_gem` call happens, it would first remote the `spec` and attached `Bundler::Source::Git` from the group before adding the new one.  If it was in reverse, it would then nothing would get removed, and the new git source would be added.

So if any preceding `git` based source will be removed that matches the removed `spec` when `override_gem` is called.  Basically what happens in my example above, where only `github.com/NickLaMuro/manageiq-api` would win out.